### PR TITLE
fix(lint): fixes golang linter errors

### PR DESCRIPTION
**Pull request by fzipi**
Signed-off-by: Felipe Zipitria <felipe.zipitria@owasp.org>

Fixes many errors from golang linter. Plenty more to continue.

### CODE REVIEW

**Review by jptosso — CHANGES_REQUESTED** [review]
```func (br *BodyBuffer) Reader() io.Reader {``` to io.Reader, error is version breaking

package log can be replaced with ```tools.Logger```

File: seclang/directives.go [hunk 1]
```diff
@@ -40,18 +41,30 @@ func directiveSecMarker(p *Parser, opts string) error {
...
 	p.Waf.Logger.Debug("added secmark rule")
 	return nil
 }
 
 func directiveSecAction(p *Parser, opts string) error {
 	rule, err := p.ParseRule(opts, false)
 	if err != nil {
-		p.log(fmt.Sprintf("Failed to compile rule (%s): %s", err, opts))
+		if perr := p.log(fmt.Sprintf("Failed to compile rule (%s): %s", err, opts)); perr != nil {
+			log.Fatal(perr) // can't write to log, fatal
```
[the remark this answers is not in the document]
Comment (fzipi) [reply seclang/directives.go:58]: Sure, we can just return that error. Changed.

File: transformations/md5.go [hunk 2]
```diff
@@ -17,11 +17,14 @@ package transformations
...
 	"io"
+	"log"
 )
 
 func Md5(data string, utils *Tools) string {
 	h := md5.New()
-	io.WriteString(h, data)
+	_, err := io.WriteString(h, data)
+	if err != nil {
+		log.Default().Println("error writing md5 string")
```
[the remark this answers is not in the document]
Comment (fzipi) [reply transformations/md5.go:27]: Good to know, changed.

File: transformations/sha1.go [hunk 3]
```diff
@@ -17,11 +17,14 @@ package transformations
...
 	"io"
+	"log"
 )
 
 func Sha1(data string, utils *Tools) string {
 	h := sha1.New()
-	io.WriteString(h, data)
+	_, err := io.WriteString(h, data)
+	if err != nil {
+		log.Default().Println("error writing sha1 string")
```
[the remark this answers is not in the document]
Comment (fzipi) [reply transformations/sha1.go:27]: Changed.

File: body_buffer.go [hunk 4]
```diff
@@ -53,12 +55,14 @@ func (br *BodyBuffer) Write(data []byte) (n int, err error) {
 }
 
 // Reader Returns a working reader for the body buffer in memory or file
-func (br *BodyBuffer) Reader() io.Reader {
+func (br *BodyBuffer) Reader() (r io.Reader, err error) {
```
[the remark this answers is not in the document]
Comment (fzipi) [reply body_buffer.go:58]: Reverting.

**Comment by jptosso** [comment]
Many linter fixes

**Pull request merged**